src/posix/platform/netif.cpp

```diff
@@ -565,18 +565,19 @@ static void processReceive(otMessage *aMessage, void *aContext)
 
 static void processTransmit(otInstance *aInstance)
 {
-    otMessage *message = nullptr;
-    ssize_t    rval;
-    char       packet[kMaxIp6Size];
-    otError    error  = OT_ERROR_NONE;
-    size_t     offset = 0;
+    otMessage *       message = nullptr;
+    ssize_t           rval;
+    char              packet[kMaxIp6Size];
+    otError           error  = OT_ERROR_NONE;
+    size_t            offset = 0;
+    otMessageSettings settings{otThreadGetDeviceRole(aInstance) != OT_DEVICE_ROLE_DISABLED, OT_MESSAGE_PRIORITY_LOW};
```

Comment:
Just a suggestion here which I think help with readability of the code (I am fine with it as is).

Instead of initializing the `otMessageSetting` struct when we declare the variable, how about if we delare the variable and then init its fields later when we use it:
```
    otMessageSettings settings;
```
And before `message = otIp6NewMessage(aInstance, &settings)` statement:
```
    setting.mLinkSecurityEnabled = (otThreadGetDeviceRole(aInstance) != OT_DEVICE_ROLE_DISABLED);
    setting.mPriority            = OT_MESSAGE_PRIORITY_LOW;
```